Decprecating routed decoder

ejguan opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#70990 Decprecating routed decoder**

Releasing `decode` API for domains to let them implement custom `decode` DataPipe for now.

Differential Revision: [D33477620](https://our.internmc.facebook.com/intern/diff/D33477620)

ejguan committed:
Decprecating routed decoder

[ghstack-poisoned]

ejguan commented:
@ejguan has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33477620).

NivekT commented on torch/utils/data/datapipes/iter/routeddecoder.py:
```diff
@@ -10,7 +11,7 @@
 
 
-@functional_datapipe('decode')
+@functional_datapipe('routed_decode')
```
Is there anyone using `.decode` before? Will this break anything for users?

ejguan replied:
I am removing it because no one from domains really uses it 😢

NivekT reviewed (approved):
Looks like some CIs are breaking, not sure if related, but otherwise LGTM!

ejguan committed:
Update on "Decprecating routed decoder"


Releasing `decode` API for domains to let them implement custom `decode` DataPipe for now.

Differential Revision: [D33477620](https://our.internmc.facebook.com/intern/diff/D33477620)

[ghstack-poisoned]

ejguan commented:
@ejguan has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33477620).
